gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java

```diff
@@ -140,6 +142,62 @@ public void eagerInitialization_fails_withInvalidCredentialsConfiguration() {
             "non-existent.json (The system cannot find the file specified)");
   }
 
+  @Test
+  public void read_file_throws_exception() throws IOException {
+    URI bucketPath = new Path("gs://test-file").toUri();
+    GoogleHadoopFileSystem myGhfs = new GoogleHadoopFileSystem();
+    myGhfs.initialize(bucketPath, new Configuration());
+
+    // custom channel
+    SeekableByteChannel channel =
+        new SeekableByteChannel() {
+          @Override
+          public boolean isOpen() {
+            return false;
+          }
+
+          @Override
+          public void close() throws IOException {
+            throw new IOException("close not implemented");
+          }
+
+          @Override
+          public int read(ByteBuffer buffer) throws IOException {
+            throw new IOException("read throws exception");
+          }
+
+          @Override
+          public int write(ByteBuffer byteBuffer) throws IOException {
+            throw new IOException("write not implemented");
+          }
+
+          @Override
+          public long position() throws IOException {
+            throw new IOException("position not present");
+          }
+
+          @Override
+          public SeekableByteChannel position(long l) throws IOException {
+            throw new IOException("set position not implemented");
+          }
+
+          @Override
+          public long size() throws IOException {
+            throw new IOException("size not present");
+          }
+
+          @Override
+          public SeekableByteChannel truncate(long l) throws IOException {
+            throw new IOException("truncate not implemented");
+          }
+        };
+
+    GoogleHadoopFSInputStream inputStream =
+        new GoogleHadoopFSInputStream(myGhfs, bucketPath, channel, null);
+    assertThrows(
+        "read throws exception", IOException.class, () -> inputStream.read(new byte[1], 0, 1));
```

Comment:
should we verify single byte read api as well?

Reply:
added test